pkg/authentication/firebase.go

```diff
@@ -44,3 +45,26 @@ func NewFirebaseWithTokenVerifier(firebaseAuth FirebaseTokenVerifier) Authentica
 		firebaseAuth: firebaseAuth,
 	}
 }
+
+// firebaseRefresher uses the firebase application to refresh the keys used to verify the token signature.
+// The firebase.App has an internal mechanism to avoid repeating this operation for every request.
+type firebaseRefresher struct {
+	app *firebase.App
+}
```

Comment:
Is this meant to be passed to `firebaseAuthentication` as its `FirebaseTokenVerifier`?

Something like this

```go
fbApp := firebase.NewApp([...])
fbVerifier := NewFirebaseRefresher(app)
authN := NewFirebaseWithTokenVerifier(fbVerifier)

[...]
```

Can you please document the intended use or interaction?

Reply:
I've added an example to the `NewFirebaseRefresher` func.